frontend/src/components/Final/Final.test.js

```diff
@@ -0,0 +1,115 @@
+import { BrowserRouter, Router } from 'react-router-dom';
+import { createMemoryHistory } from 'history'
+
+import Final from './Final'; // Adjust the import path as necessary
+
+
+jest.mock('../../API', () => ({
+    finalizeSession: jest.fn(),
+}));
+
+jest.mock('../../config', () => ({
+    URLS: {
+        AMLHome: '/aml',
+        profile: '/profile',
+    },
+}));
+
+describe('Final Component', () => {
+    it('renders correctly with given props', () => {
+        render(
+            <BrowserRouter>
+                <Final
+                    experiment={{ slug: 'test-experiment' }}
+                    participant="participant-id"
+                    session="session-id"
+                    score={100}
+                    final_text="<p>Final Text</p>"
+                    rank={1}
+                />
+            </BrowserRouter>
+        );
+
+        expect(screen.getByText(/Final Text/i)).toBeInTheDocument();
+        expect(screen.getByTestId('score')).toBeInTheDocument(); // Adjust based on how you display points
+    });
+
+    it('calls onNext prop when button is clicked', async () => {
+        const onNextMock = jest.fn();
+        render(
+            <BrowserRouter>
+                <Final
+                    button={{ text: 'Next', link: '' }}
+                    onNext={onNextMock}
+                />
+            </BrowserRouter>
+        );
+
+        fireEvent.click(screen.getByText('Next'));
+        await waitFor(() => {
+            expect(onNextMock).toHaveBeenCalled();
+        });
+    });
+
+    it('does not render rank and social components when props are not provided', () => {
+        render(
+            <BrowserRouter>
+                <Final
+                    experiment={{ slug: 'test-experiment' }}
+                    participant="participant-id"
+                    session="session-id"
+                    score={100}
+                    final_text="<p>Final Text</p>"
+                />
+            </BrowserRouter>
+        );
+
+        expect(screen.queryByText('Rank')).not.toBeInTheDocument();
+        expect(screen.queryByText('Social')).not.toBeInTheDocument();
+    });
+
+    it('navigates to profile page when profile link is clicked', async () => {
+
+        const history = createMemoryHistory();
+
+        const mockActionTexts = {
+            all_experiments: 'All Experiments',
+            profile: 'Profile',
+        };
+
+        render(
+            <Router history={history}>
+                <Final
+                    show_profile_link={true}
+                    action_texts={mockActionTexts}
+                />
+            </Router>
+        );
+
+        const profileLink = screen.getByTestId('profile-link');
+
+        expect(profileLink).toBeInTheDocument();
+
+        fireEvent.click(profileLink)
+
+        expect(history.length).toBe(2);
+        expect(history.location.pathname).toBe('/profile');
```

Comment:
Ah, I see: you check whether the profile link correctly forwards to the desired location. Very nice! The test stumbles over the import from `history`, however - does this need to be added to the dependencies in package.json?